app/controllers/ticket_purchases_controller.rb

```diff
@@ -8,23 +8,42 @@ class TicketPurchasesController < ApplicationController
 
   def create
     current_user.ticket_purchases.by_conference(@conference).unpaid.destroy_all
+
+    # Create a ticket purchase which can be paid or unpaid
+    count_registration_tickets_before = current_user.count_registration_tickets(@conference)
     message = TicketPurchase.purchase(@conference, current_user, params[:tickets].try(:first))
-    if message.blank?
-      if current_user.ticket_purchases.by_conference(@conference).unpaid.any?
-        redirect_to new_conference_payment_path,
-                    notice: 'Please pay here to get tickets.'
-      elsif current_user.ticket_purchases.by_conference(@conference).paid.any?
-        redirect_to conference_physical_tickets_path,
-                    notice: 'You already have tickets for the conference.'
-      elsif @conference.tickets.for_registration.any?
-        redirect_to conference_tickets_path(@conference.short_title),
-                    error: 'Please get at least one ticket to continue.'
-      else
-        redirect_to conference_conference_registration_path(@conference.short_title)
-      end
-    else
+    # The new ticket_purchase has been added to the database. current_user.ticket_purchases contains the new one.
+    count_registration_tickets_after = current_user.count_registration_tickets(@conference)
+
+    # Failed to create ticket purchase
+    unless message.blank?
       redirect_to conference_tickets_path(@conference.short_title),
                   error: "Oops, something went wrong with your purchase! #{message}"
+      return
+    end
+
+    # Current user already paid for a registration ticket and the current ticket purchase contains one
+    if count_registration_tickets_before == 1 && count_registration_tickets_after > 1
```

Comment:
count_registration_tickets_before and _after is only used twice in this function (line 26 and line 41) and its called for verifications. It might be cleaner to separate them into their own function like check_before_and_after and then call that from within def create.